cmd/egress-v4-cni-plugin/snat/snat.go

```diff
@@ -0,0 +1,127 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package snat
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/coreos/go-iptables/iptables"
+)
+
+func iptRules4(target, src net.IP, chain, comment string, useRandomFully bool) [][]string {
+	var rules [][]string
+
+	// Accept/ignore multicast (just because we can)
+	rules = append(rules, []string{chain, "-d", "224.0.0.0/4", "-j", "ACCEPT", "-m", "comment", "--comment", comment})
+
+	// SNAT
+	args := []string{
+		chain,
+		"-j", "SNAT",
+		"--to-source", target.String(),
+		"-m", "comment", "--comment", comment,
+	}
+	if useRandomFully {
+		args = append(args, "--random-fully")
+	}
+	rules = append(rules, args)
+	rules = append(rules, []string{"POSTROUTING", "-s", src.String(), "-j", chain, "-m", "comment", "--comment", comment})
+
+	return rules
+}
+
+//Setup a rule to block egress traffic directed to 169.254.172.0/22 from the Pod
+func SetupRuleToBlockNodeLocalV4Access() error{
+	ipt, _ := iptables.NewWithProtocol(iptables.ProtocolIPv4)
+	if err := ipt.AppendUnique("filter", "OUTPUT", "-d", "169.254.172.0/22", "-m", "comment",
+			"--comment", "Block Node Local Pod access via IPv4", "-j", "DROP"); err != nil {
+			return fmt.Errorf("failed adding v4 drop route: %v", err)
+			}
```

Comment:
nit: the indenting on this close brace is odd - is this what `go fmt` produced, or does this indicate that this file has not been `go fmt`'ed?  (If the latter, we should think about adding an automated PR test to verify this)

Reply:
Yeah, missed `go fmt` on this file. We do have `fmt` check as part of the CI pipeline and that failed for this PR as well. Plan was to address it as part of the CR comments.